services/leader/service.go

```diff
@@ -0,0 +1,189 @@
+package leader
+
+import (
+	"errors"
+	"fmt"
+	"sync"
+	"sync/atomic"
+
+	"github.com/coreos/etcd/clientv3"
+	"github.com/m3db/m3cluster/services"
+	"github.com/m3db/m3cluster/services/leader/campaign"
+	"github.com/m3db/m3x/errors"
+)
+
+const (
+	leaderKeyPrefix = "_ld"
+	keySeparator    = "/"
+	keyFormat       = "%s/%s"
+)
+
+var (
+	// errClientClosed indicates the election service client has been closed and
+	// no more elections can be started.
+	errClientClosed = errors.New("election client is closed")
+)
+
+type multiClient struct {
+	sync.RWMutex
+
+	closed     uint32
+	clients    map[string]*clientEntry
+	opts       Options
+	etcdClient *clientv3.Client
+}
+
+// clientEntry stores a cached client as well as the TTL it was created with so
+// that a user will receive an error if they try to create a new client with a
+// different TTL
+type clientEntry struct {
+	client *client
+	ttl    int
```

Comment:
I think it makes more sense to keep it as an `int` since that's the type we pass to [`concurrency.WithTTL`](https://godoc.org/github.com/coreos/etcd/clientv3/concurrency#WithTTL).